dotnet/src/Connectors/Connectors.UnitTests/OpenAI/AudioToText/OpenAIAudioToTextServiceTests.cs

```diff
@@ -59,6 +59,21 @@ public void ConstructorWithOpenAIClientWorksCorrectly(bool includeLoggerFactory)
         Assert.Equal("model-id", service.Attributes["ModelId"]);
     }
 
+    [Theory]
+    [InlineData(true)]
+    [InlineData(false)]
+    public void ConstructorWithCustomEndpointWorksCorrectly(bool includeLoggerFactory)
```

Comment:
Add UnitTests for:
- Ensures that the verification is throwing the exception when missing ModelId or Endpoint
- Ensure that the endpoint is working when the httpClient has a baseAddress defined.